pkg/maglev/maglev.go

```diff
@@ -16,55 +16,46 @@ package maglev
 
 import (
 	"encoding/base64"
+	"fmt"
 
-	"github.com/cilium/cilium/pkg/logging"
-	"github.com/cilium/cilium/pkg/logging/logfields"
-	"github.com/sirupsen/logrus"
-	"github.com/spaolacci/murmur3"
+	"github.com/cilium/cilium/pkg/murmur3"
 )
 
 const (
-	subsystem = "maglev"
-
 	DefaultTableSize = 16381
 
-	// seed=$(head -c16 /dev/urandom | base64 -w0)
-	DefaultHashSeed = "FYJf2hrSynYxcZrkMWTGYA=="
+	// seed=$(head -c12 /dev/urandom | base64 -w0)
+	DefaultHashSeed = "JLfvgnHc2kaSUFaI"
 )
 
 var (
-	seedMurmur0 uint32
-	seedMurmur1 uint32
+	seedMurmur uint32
 
 	SeedJhash0 uint32
 	SeedJhash1 uint32
 )
 
-var log = logging.DefaultLogger.WithField(logfields.LogSubsys, subsystem)
-
-func InitMaglevSeeds(seed string) {
+func InitMaglevSeeds(seed string) error {
 	d, err := base64.StdEncoding.DecodeString(seed)
 	if err != nil {
-		log.WithError(err).WithFields(logrus.Fields{
-			logfields.HashSeed: seed,
-		}).Fatal("Cannot decode base64 Maglev hash seed.")
+		return fmt.Errorf("Cannot decode base64 Maglev hash seed %q: %s", seed, err)
```

Comment:
Use `%w` to wrap the underlying error:

```suggestion
		return fmt.Errorf("Cannot decode base64 Maglev hash seed %q: %w", seed, err)
```